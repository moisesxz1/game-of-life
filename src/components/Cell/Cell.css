.cell {
    width: 20px;
    height: 20px;
    border: 1px solid #ccc;
}

.is-alive {
    background-color: black;
}