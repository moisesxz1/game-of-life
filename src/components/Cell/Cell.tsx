import React from "react";
import "./Cell.css";

const Cell: React.FC<{
  isAlive: boolean;
  onClick: () => void;
}> = React.memo(({ isAlive, onClick }) => {
  return (
    <div
      data-testid="cell"
      className={`cell ${isAlive ? "is-alive" : "is-dead"}`}
      onClick={onClick}
    />
  );
});

export default Cell;
