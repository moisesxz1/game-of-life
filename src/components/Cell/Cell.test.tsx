import React from "react";
import { render } from "@testing-library/react";
import Cell from "./Cell";

const mockedOnClickFunction = jest.fn();

describe("Cell Component", () => {
  it("Matches the snapshot", () => {
    const { asFragment } = render(
      <Cell isAlive={false} onClick={mockedOnClickFunction} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Applies is-alive class when isAlive is true", () => {
    const { container } = render(<Cell isAlive={true} onClick={() => {}} />);
    expect(container.firstChild).toHaveClass("is-alive");
  });

  test("Calls onClick when clicked", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Cell isAlive={false} onClick={handleClick} />
    );

    const cell = getByTestId("cell");
    cell.click();

    expect(handleClick).toHaveBeenCalled();
  });

  test("matches snapshot when dead", () => {
    const { asFragment } = render(<Cell isAlive={false} onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
