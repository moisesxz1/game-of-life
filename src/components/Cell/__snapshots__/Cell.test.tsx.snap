// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="cell is-dead"
    data-testid="cell"
  />
</DocumentFragment>
`;

exports[`Cell Component matches snapshot when dead 1`] = `
<DocumentFragment>
  <div
    class="cell is-dead"
    data-testid="cell"
  />
</DocumentFragment>
`;
